### DISCUSSION TITLE
OpenLiteSpeed and LiteSpeed Support

### DISCUSSION THREAD

**Discussion by edavrio** [DirectoryLister/DirectoryLister#584] [Ideas & Features]
This is on of a kind piece of software. It's very light, very fast and very nice looking.
I think the biggest improve this app can get is have a bolt on integration for LiteSpeed.
This is the default autoindex used by OLS :[https://github.com/litespeedtech/openlitespeed/blob/master/dist/share/autoindex/default.php](https://github.com/litespeedtech/openlitespeed/blob/master/dist/share/autoindex/default.php)

So having this perk in OLS will boost the app and will be the number one alternative.
LiteSpeed is the fastest web server for serving static and php files with embedded Cache engine.
It outperforms any webserver in any challenge.

**Comment by PHLAK** [comment]
Thanks, glad you're enjoying Directory Lister!

What exactly are you looking for when you say "bolt on integration for LiteSpeed"?

**Reply by edavrio** [reply]
Well, Well if the DirectoryLister would support LS_AI_PATH variable which is passed by the webserver, then it could easy replace default autoindex which looks like apache mostly ) Take a look here [https://mirror.ihost.md/alpine/](https://mirror.ihost.md/alpine/).
In both OpenLiteSpeed and LiteSpeed Enterprise there is a option to use custom php file as autoindex.
Here is the official documentation [https://openlitespeed.org/kb/enable-and-configure-auto-index/](https://openlitespeed.org/kb/enable-and-configure-auto-index/)

And in .env files a variable could turn the support on or off.

**Reply by PHLAK** [reply]
After glancing over those links (and a few others I found) I'm not sure this is possible without some refactoring. That being said I'll have to look into it further when I have more time to really understand how this works.
